lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -2156,35 +2561,39 @@ class MongoClientInterface {
                             return next(errors.InternalError);
                         }
                         bucketCount++;
-                        bucketInfos.push(bucketInfo);
+                        bucketInfos!.push(bucketInfo!);
                         return next();
                     });
             }, err => {
                 if (err) {
-                    return cb(err);
+                    return cb(err as ArsenalError);
                 }
                 return cb(null, {
                     bucketCount,
                     bucketInfos,
                 });
             })).catch(err => {
-            log.error('could not get list of collections', {
-                method: '_getBucketInfos',
-                error: err,
+                log.error('could not get list of collections', {
+                    method: '_getBucketInfos',
+                    error: err,
+                });
+                return cb(err);
```

Comment:
this one may not be an ArsenalError, e.g. in case `listCollections` fails?

Reply:
I updated to return the arsenal error if the error is of this instance, otherwise, InternalError.